pre_commit/commands/run.py

```diff
@@ -104,9 +106,19 @@ def _run_single_hook(hook, repo, args, write, skips=frozenset()):
 
     write(color.format_color(pass_fail, print_color, args.color) + '\n')
 
-    if (stdout or stderr) and (retcode or args.verbose):
+    if (stdout or stderr or file_modifications) and (retcode or args.verbose):
         write('hookid: {0}\n'.format(hook['id']))
         write('\n')
+
+        # Print a message if failing due to file modifications
+        if file_modifications:
+            extra_text = (
+                ' Additional output:\n' if (stdout or stderr) else ''
+            )
+
+            write('Files were modified by this hook.{0}'.format(extra_text))
```

Comment:
I agree that just writing a line is nicer than formatting, but the double linebreak was actually deliberate to ensure that if we have additional output, we have the same spacing at the end of the hook's full output regardless of whether `stdout` or `stderr` exist:

```
eslint......................................................................................Failed
hookid: hook

Files were modified by this hook. Additional output:

Modified foo.py!

```

and for no `stdout` or `stderr`:

```
eslint......................................................................................Failed
hookid: hook

Files were modified by this hook.

```

That's unfortunately why the code is a little weird, since without the newline stuff we'd have two linebreaks at the end of the second case (since the final `write('\n')` is not surrounded by a check for `stdout` or `stderr`). Let me know if I'm missing a better way to avoid that.